Version checker causes stack overflow when there's no internet connection

Partial stack trace below:

```
Linking module 'com.gwtplatform.mvp.MvpGwtTest.JUnit'
[INFO]    Invoking Linker Verify the availability of a more recent version of GWTP.
[INFO]       Checking version information for gwtp-mvp-client
[INFO]          [ERROR] Exception caught
[INFO] java.net.UnknownHostException: arcbees-stats-service.appspot.com
[INFO]  at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178)
[INFO]  at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
[INFO]  at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
[INFO]  at java.net.Socket.connect(Socket.java:579)
[INFO]  at java.net.Socket.connect(Socket.java:528)
[INFO]  at java.net.Socket.<init>(Socket.java:425)
[INFO]  at java.net.Socket.<init>(Socket.java:208)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.fetchArtifactVersion(VersionInspectorLinker.java:178)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersion(VersionInspectorLinker.java:154)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersionIfPresent(VersionInspectorLinker.java:144)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersions(VersionInspectorLinker.java:132)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.link(VersionInspectorLinker.java:97)
[INFO]  at com.google.gwt.core.ext.linker.impl.StandardLinkerContext.invokeFinalLink(StandardLinkerContext.java:328)
[INFO]  at com.google.gwt.dev.DevModeBase.link(DevModeBase.java:1009)
[INFO]  at com.google.gwt.junit.JUnitShell.maybeCompileForWebMode(JUnitShell.java:1172)
[INFO]  at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl2(CompileStrategy.java:180)
[INFO]  at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl(CompileStrategy.java:112)
[INFO]  at com.google.gwt.junit.SimpleCompileStrategy.maybeCompileModule(SimpleCompileStrategy.java:36)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1342)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1311)
[INFO]  at com.google.gwt.junit.JUnitShell.runTest(JUnitShell.java:705)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.runTest(GWTTestCase.java:421)
[INFO]  at junit.framework.TestCase.runBare(TestCase.java:141)
[INFO]  at junit.framework.TestResult$1.protect(TestResult.java:122)
[INFO]  at junit.framework.TestResult.runProtected(TestResult.java:142)
[INFO]  at junit.framework.TestResult.run(TestResult.java:125)
[INFO]  at junit.framework.TestCase.run(TestCase.java:129)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.run(GWTTestCase.java:247)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.doRun(MavenTestRunner.java:105)
[INFO]  at junit.textui.TestRunner.start(TestRunner.java:183)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.main(MavenTestRunner.java:63)
[INFO]       Checking version information for gwtp-dispatch-rpc-client
[INFO]          [ERROR] Exception caught
[INFO] java.net.UnknownHostException: arcbees-stats-service.appspot.com
[INFO]  at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178)
[INFO]  at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
[INFO]  at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
[INFO]  at java.net.Socket.connect(Socket.java:579)
[INFO]  at java.net.Socket.connect(Socket.java:528)
[INFO]  at java.net.Socket.<init>(Socket.java:425)
[INFO]  at java.net.Socket.<init>(Socket.java:208)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.fetchArtifactVersion(VersionInspectorLinker.java:178)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersion(VersionInspectorLinker.java:154)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersionIfPresent(VersionInspectorLinker.java:144)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.checkLatestVersions(VersionInspectorLinker.java:133)
[INFO]  at com.gwtplatform.common.rebind.VersionInspectorLinker.link(VersionInspectorLinker.java:97)
[INFO]  at com.google.gwt.core.ext.linker.impl.StandardLinkerContext.invokeFinalLink(StandardLinkerContext.java:328)
[INFO]  at com.google.gwt.dev.DevModeBase.link(DevModeBase.java:1009)
[INFO]  at com.google.gwt.junit.JUnitShell.maybeCompileForWebMode(JUnitShell.java:1172)
[INFO]  at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl2(CompileStrategy.java:180)
[INFO]  at com.google.gwt.junit.CompileStrategy.maybeCompileModuleImpl(CompileStrategy.java:112)
[INFO]  at com.google.gwt.junit.SimpleCompileStrategy.maybeCompileModule(SimpleCompileStrategy.java:36)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1342)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1311)
[INFO]  at com.google.gwt.junit.JUnitShell.runTest(JUnitShell.java:705)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.runTest(GWTTestCase.java:421)
[INFO]  at junit.framework.TestCase.runBare(TestCase.java:141)
[INFO]  at junit.framework.TestResult$1.protect(TestResult.java:122)
[INFO]  at junit.framework.TestResult.runProtected(TestResult.java:142)
[INFO]  at junit.framework.TestResult.run(TestResult.java:125)
[INFO]  at junit.framework.TestCase.run(TestCase.java:129)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.run(GWTTestCase.java:247)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.doRun(MavenTestRunner.java:105)
[INFO]  at junit.textui.TestRunner.start(TestRunner.java:183)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.main(MavenTestRunner.java:63)
[INFO] [ERROR] Failure in unit cache map load.
[INFO] java.util.concurrent.ExecutionException: java.lang.StackOverflowError
[INFO]  at java.util.concurrent.FutureTask.report(FutureTask.java:122)
[INFO]  at java.util.concurrent.FutureTask.get(FutureTask.java:188)
[INFO]  at com.google.gwt.dev.javac.PersistentUnitCache.awaitUnitCacheMapLoad(PersistentUnitCache.java:512)
[INFO]  at com.google.gwt.dev.javac.PersistentUnitCache.find(PersistentUnitCache.java:418)
[INFO]  at com.google.gwt.dev.javac.CompilationStateBuilder.doBuildFrom(CompilationStateBuilder.java:483)
[INFO]  at com.google.gwt.dev.javac.CompilationStateBuilder.buildFrom(CompilationStateBuilder.java:434)
[INFO]  at com.google.gwt.dev.javac.CompilationStateBuilder.buildFrom(CompilationStateBuilder.java:420)
[INFO]  at com.google.gwt.dev.cfg.ModuleDef.getCompilationState(ModuleDef.java:495)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1345)
[INFO]  at com.google.gwt.junit.JUnitShell.runTestImpl(JUnitShell.java:1311)
[INFO]  at com.google.gwt.junit.JUnitShell.runTest(JUnitShell.java:705)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.runTest(GWTTestCase.java:421)
[INFO]  at junit.framework.TestCase.runBare(TestCase.java:141)
[INFO]  at junit.framework.TestResult$1.protect(TestResult.java:122)
[INFO]  at junit.framework.TestResult.runProtected(TestResult.java:142)
[INFO]  at junit.framework.TestResult.run(TestResult.java:125)
[INFO]  at junit.framework.TestCase.run(TestCase.java:129)
[INFO]  at com.google.gwt.junit.client.GWTTestCase.run(GWTTestCase.java:247)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:255)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:250)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.doRun(MavenTestRunner.java:105)
[INFO]  at junit.textui.TestRunner.start(TestRunner.java:183)
[INFO]  at org.codehaus.mojo.gwt.test.MavenTestRunner.main(MavenTestRunner.java:63)
```

--- a/gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/rebind/VersionInspectorLinker.java
+++ b/gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/rebind/VersionInspectorLinker.java
@@ -41,7 +41,6 @@ import com.google.gwt.core.ext.linker.SelectionProperty;
 import com.google.gwt.core.ext.linker.Shardable;
 
 import static com.google.gwt.core.ext.TreeLogger.Type.DEBUG;
-import static com.google.gwt.core.ext.TreeLogger.Type.ERROR;
 
 @Shardable
 @LinkerOrder(Order.POST)
@@ -93,7 +92,7 @@ public class VersionInspectorLinker extends Linker {
     @Override
     public ArtifactSet link(TreeLogger logger, LinkerContext context, ArtifactSet artifacts, boolean onePermutation)
             throws UnableToCompleteException {
-        if (!onePermutation && !isSuperDevMode(context) && canVerifyNewerVersion(context)) {
+        if (!onePermutation && !isSuperDevMode(context) && !isTest(context) && canVerifyNewerVersion(context)) {
             checkLatestVersions(logger);
         }
 
@@ -111,6 +110,22 @@ public class VersionInspectorLinker extends Linker {
         }
 
         return isSdm;
+    }
+
+    private boolean isTest(LinkerContext context) {
+        boolean isTest = false;
+
+        for (ConfigurationProperty property : context.getConfigurationProperties()) {
+            String name = property.getName();
+            List<String> values = property.getValues();
+
+            if ("junit.moduleName".equals(name) && !values.isEmpty() && !values.get(0).isEmpty()) {
+                isTest = true;
+                break;
+            }
+        }
+
+        return isTest;
     }
 
     private boolean canVerifyNewerVersion(LinkerContext context) {
@@ -161,7 +176,7 @@ public class VersionInspectorLinker extends Linker {
                 warnVersion(groupId, artifactId, latestVersion, currentVersion);
             }
         } catch (Exception e) {
-            logger.getTreeLogger().log(ERROR, "Exception caught", e);
+            logger.getTreeLogger().log(DEBUG, "Exception caught", e);
         }
     }
 

